x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -774,118 +775,98 @@ public PlanFactory visitRerankCommand(EsqlBaseParser.RerankCommandContext ctx) {
-
-        return visitInferenceId(expression(ctx.parameter()));
-    }
+    private Completion applyCompletionOptions(Completion completion, EsqlBaseParser.CommandNamedParametersContext ctx) {
+        MapExpression optionsExpresion = visitCommandNamedParameters(ctx);
 
-    private Literal visitInferenceId(EsqlBaseParser.InferenceCommandOptionValueContext ctx) {
-        if (ctx.identifier() != null) {
-            return Literal.keyword(source(ctx), visitIdentifier(ctx.identifier()));
+        if (optionsExpresion == null || optionsExpresion.containsKey(Completion.INFERENCE_ID_OPTION_NAME) == false) {
+            // Having a mandatory named parameter for inference_id is an antipattern, but it will be optional in the future when we have a
+            // default LLM. It is better to keep inference_id as a named parameter and relax the syntax when it will become optional than
+            // completely change the syntax in the future.
+            throw new ParsingException(source(ctx), "Missing mandatory option [{}] in COMPLETION", Completion.INFERENCE_ID_OPTION_NAME);
         }
 
-        return visitInferenceId(expression(ctx.constant()));
-    }
+        Map<String, Expression> optionsMap = visitCommandNamedParameters(ctx).keyFoldedMap();
 
-    private Literal visitInferenceId(Expression expression) {
-        if (expression instanceof Literal literal) {
-            if (literal.value() == null) {
+        Expression inferenceId = optionsMap.remove(Completion.INFERENCE_ID_OPTION_NAME);
+        if (inferenceId != null) {
+            if (inferenceId instanceof Literal inferenceIdLiteral && DataType.isString(inferenceId.dataType())) {
+                if (inferenceIdLiteral.value() == null) {
+                    throw new ParsingException(
+                        inferenceId.source(),
+                        "[{}] option cannot be null or undefined in COMPLETION",
+                        Completion.INFERENCE_ID_OPTION_NAME
+                    );
+                }
+                completion = completion.withInferenceId(inferenceId);
+            } else {
                 throw new ParsingException(
-                    expression.source(),
-                    "Parameter [{}] is null or undefined and cannot be used as inference id",
-                    expression.source().text()
+                    inferenceId.source(),
+                    "Option [{}] only support string in COMPLETION but [{}] cannot be used as string",
+                    Completion.INFERENCE_ID_OPTION_NAME
                 );
             }
+        }
 
-            return literal;
-        } else if (expression instanceof UnresolvedAttribute attribute) {
-            // Support for unquoted inference id
-            return new Literal(expression.source(), attribute.name(), KEYWORD);
+        if (optionsMap.isEmpty() == false) {
+            throw new ParsingException(
+                source(ctx),
+                "Unknown option [{}] in Completion command",
```

Comment:
nitpicking again, but for functions that expect options and we use an unknown option we return something like this:
 
 ```
 line 3:9: Invalid option [foo] in [match(description, "aa", { "foo": "abc"})], expected one of [prefix_length, lenient, zero_terms_query, analyzer, fuzzy_transpositions, fuzzy_rewrite, fuzziness, max_expansions, boost, operator, auto_generate_synonyms_phrase_query, minimum_should_match]"""
 ```
 
 this one I think we can address separately and not in this PR.

Reply:
Done!